src/aws_encryption_sdk/keyring/base.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Base class interface for Keyrings."""
+
+
+class Keyring(object):
+    """Parent interface for Keyring classes."""
+
+    def on_encrypt(self, encryption_materials):
+        """Generates a data key and encrypts it using all wrapping keys the Keyring is associated with.
```

Comment:
Looks like I still have this pylint check disabled, but please make all docstring descriptions in imperative mood.

ex:
```suggestion
        """Generate a data key if not present and encrypt it using any available wrapping keys.
```

Reply:
I will do that